api/src/services/accesscontrols.js

```diff
@@ -20,6 +22,15 @@ const grantsObject = {
       'update:any': ['*'],
       'delete:any': ['*'],
     },
+    dataset_name: {
+      'read:any': ['*'],
```

Comment:
Expose Dataset Name as a resource to all 3 roles, so they can verify if a dataset with a selected name exists.